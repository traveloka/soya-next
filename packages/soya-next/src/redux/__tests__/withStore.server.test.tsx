/**
 * @jest-environment node
 */
import setupTest from "./utils/withStoreSetup";

describe("withStore - server", () => {
  it("should add store to page props", async () => {
    const { PageWithStore } = setupTest();
    const props = await PageWithStore.getInitialProps({} as any);
    expect(props).toHaveProperty("init");
    expect(props).toHaveProperty("store");
    expect(props).toMatchSnapshot();
  });
});
