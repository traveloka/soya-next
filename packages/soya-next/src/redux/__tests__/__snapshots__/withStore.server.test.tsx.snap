// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withStore - server should add store to page props 1`] = `
Object {
  "init": true,
  "reduxState": Object {},
  "store": Object {
    "@@observable": [Function],
    "addReducer": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "soya": true,
    "subscribe": [Function],
  },
}
`;
